### ISSUE TITLE
Using CSLA 2019: CSLA .NET Overview ebook online

### ISSUE THREAD

**Issue by rockfordlhotka** [MarimerLLC/cslaforum#706]
Hello CSLA .NET community!

I have decided to put the new [Using CSLA 2019: CSLA .NET Overview](https://store.lhotka.net/using-csla-2019-csla-net-overview) ebook online **for free**.

As I complete rewriting the remainder of the book series they'll be available for purchase, but I've decided there's value to the community and the framework to have this overview available to everyone.

I'm still distributing the ebook via the store, but without cost - so please enjoy!

(and let me know if you find any errors in the book - there's always room for improvement)

**Comment by itgoran** [comment]
@rockfordlhotka 
I would like to suggest that you create a dedicated repository (announcements) which can serve this purposes. Although I'm subscribed to this notification,  I was thinking that a dedicated repository can server all kinds of announcements and will not be overwhelmed with forum issues. 
Some of the announcements could be:

- A new book (like this one)
- Any conference at which you attend to and community can see you
- New releases (minor, major)
- Security patches (if any)
- Workshop if any
- Podcast 

What do you think?

**Comment by Chicagoan2016** [comment]
@rockfordlhotka, thank you sir, just downloaded and going to read it.

Kind regards

**Comment by Chicagoan2016** [comment]
@rockfordlhotka , I have a quick observation, not trying to be a stickler :), I was reading the book and I read this on p-18
 

> A later book in this book series will discuss the use of service-oriented architectures with CSLA .NET in detail.

> 
I think a similar statement was mentioned in Using Csla 4 books too :).
I would love to see CSLA with SOA, I remember there was  a chapter 'WCF service Interface' in Expert C# 2008 business objects' book.

Kind regards

**Comment by rockfordlhotka** [comment]
@Chicagoan2016 yes, my original intent was to write such a book, but the whole "SOA" hype wave kind of crashed back then. Now "microservices" is the new name and has hype, so I really do intend on writing that service-oriented book this time 😄

**Comment by Chicagoan2016** [comment]
@rockfordlhotka  thank you,
 I have another question : ).  I remember from reading your books and listening to your talks ( there used to be Magenic University), you were of the opinion that a 'webservice' is essentially another UI like ASP.NET , Winforms, WPF etc. that talks to Csla business objects. Could we still consider the same about microservices?

kind regards

**Comment by rockfordlhotka** [comment]
Yes, absolutely. _Especially_ with microservices, where each service is, by definition, a standalone unit of functionality (aka app). A microservice very clearly has an interface that's defined by the API it exposes and the messages it accepts/returns. That interface is most often a combination of HTTP verbs and JSON.

Behind that interface is Interface Control code - your actual service implementation - typically a controller.

Behind that is your business layer - ideally (😄 ) CSLA-based objects.

**Comment by bertiefrog** [comment]
I was considering purchasing your eBooks for CSLA 4, should I wait until you've put up the set for 2019, any ideas on how long I should wait, or is it like a Brexit thing, possible delay?

**Comment by Chicagoan2016** [comment]
@bertiefrog , I was in a very similar position a few years ago, I had used Csla 1.xx and had maintained a decent sized (200k lines custom code) project in Csla.1xx and jumped straight to Csla 4.xx.
I found Using Csla 4.0 books to be extremely helpful. I read and still go back and reread those books, but whenever I needed some explanation, Rocky and the rest of Csla community have been helpful beyond words, I consider @rockfordlhotka  a teacher and a mentor and special kudos to @jonnybee .
you guys are the best!, can't thank you enough.

So YES do buy the Csla 4.0 books :)

**Comment by rockfordlhotka** [comment]
@bertiefrog, The total size of the old ebook is over 900 pages. I expect the new content to be similar or larger. Which is to say that I expect this project to take several months to complete, and the updated books will come out over a fairly long timeframe.

**Comment by itgoran** [comment]
@bertiefrog 
CSLA 4.0 is worth buying since the content basically guides any developer in the journey of developing small and large applications. 
Additionally, it does provide necessary information about CSLA internals and can be used as a reference book for, not just you, but also for your team members to become productive in the early stages in the project.

**Comment by ngm011** [comment]
Rocky is back to writing?! 

Oh my, that’s amazing news! 

I’ve got all your publications since VB6 BO’s and Distributed Objects to CSLA eBooks and videos.
You’re one of the most pragmatic authors with such a gifted writing skills.

Thank you so much for this contribution to community, I can’t wait for more stuff to come.
